bin/build-electron

```diff
@@ -56,31 +59,50 @@ while [ "$1" != "" ]; do
     shift
 done
 
-get_default_platform()
-{
-    hash rustup 2>/dev/null || { echo >&2 "Make sure you have rustup installed and properly configured! Aborting."; exit 1; }
-
-    case "$(rustup show active-toolchain)" in 
-        *"x86_64-apple-darwin"* )
-            echo "darwin"
-            ;;
-        *"x86_64-pc-windows"* )
-            echo "win32"
-            ;;
-        *"x86_64-unknown-linux"* )
-            echo "linux"
-            ;;
-        * )
-            echo "unknown"
-    esac
-}
-
-DEFAULT_PLATFORM=$(get_default_platform)
-if [ "${DEFAULT_PLATFORM}" = "unknown" ]
-then
-    echo "Unknown platform detected!\nPlease make sure you have installed a valid Rust toolchain via rustup! Aborting."
-    exit 1
-fi
+case "$TARGET_ARCH" in
+    "x64")
+        GN_ARCH=x64
+        CARGO_ARCH=x86_64
+        ;;
+    "ia32")
+        GN_ARCH=x86
+        CARGO_ARCH=i686
+        ;;
+    "arm64")
+        GN_ARCH=arm64
+        CARGO_ARCH=aarch64
+        ;;
+    *)
+        echo "Unsupported architecture"
+        exit 1
+        ;;
+esac
+
+hash rustup 2>/dev/null || { echo >&2 "Make sure you have rustup installed and properly configured! Aborting."; exit 1; }
+
+RUSTFLAGS_WIN=
+
+case "$(rustup show active-toolchain)" in 
+    *"x86_64-apple-darwin"* )
+        DEFAULT_PLATFORM="darwin"
+        CARGO_TARGET="${CARGO_ARCH}-apple-darwin"
+        ;;
+    *"x86_64-pc-windows"* )
+        DEFAULT_PLATFORM="win32"
+        CARGO_TARGET="${CARGO_ARCH}-pc-windows-msvc"
+        # Static linking to prevent build errors on Windows ia32
+        RUSTFLAGS_WIN="-C target-feature=+crt-static"
```

Comment:
Could/should this be done in the src/rust/build.rs file? There is a section in there where we force it to link to the static libc.

Reply:
@jim-signal I saw that there's already something for that in `build.rs`:

https://github.com/signalapp/ringrtc/blob/8c9259ff7d7cae08a1c1714167e7c9d5e085f3be/src/rust/build.rs#L45-L50

When I don't add `-C target-feature=+crt-static` to the build process, I get linker errors:

```
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX64\\x86\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LARGEADDRESSAWARE" "/SAFESEH" "/LIBPATH:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.0.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.1.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.10.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.11.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.12.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.13.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.14.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.15.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.2.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.3.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.4.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.5.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.6.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.7.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.8.rcgu.o" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.ringrtc.cvmzzsmy-cgu.9.rcgu.o" "/OUT:C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.dll" "/DEF:C:\\Users\\Dennis\\AppData\\Local\\Temp\\rustcvmeYrS\\lib.def" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.2b2wqrt5ttopb0e9.rcgu.o" "/OPT:REF,ICF" "/DLL" "/IMPLIB:C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\ringrtc.dll.lib" "/DEBUG" "/NATVIS:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis" "/LIBPATH:C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps" "/LIBPATH:C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\release\\deps" "/LIBPATH:../../src/webrtc/src/out/Release/obj/" "/LIBPATH:C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "webrtc.lib" "winmm.lib" "dmoguids.lib" "msdmo.lib" "wmcodecdspuuid.lib" "secur32.lib" "iphlpapi.lib" "libcmt.lib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libbase64-dec81ab446f62e09.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\liblibc-1c1e89aadc6b9fb0.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libneon-0686d6bc0dd0d6ba.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libsemver-b5ae68705386b0f0.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libsemver_parser-7c05dc7df11561ec.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcslice-ecdaae8bdff1a339.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libneon_runtime-f5a25469333a2517.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\liblibloading-d1ba61772e48b9f6.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libwinapi-8e9f49b6bd285354.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libsmallvec-fd14e9653beac749.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libtokio-29fa05a912bf2d41.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libnum_cpus-5ada646cd69bd0a3.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libpin_project_lite-4ad6b54f6b3c6b34.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libregex-7a86961ed5bfb7ef.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libthread_local-7e3b5a81c9e88937.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libregex_syntax-a7f96430cd6b6ad4.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libaho_corasick-13ce9f99e015b04f.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\liblazy_static-8adbd3d5ac795c6f.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libserde_json-fa0f04708dc4d332.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libryu-54a417814c329448.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libitoa-306b9ae59e583c94.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libserde-3e6d1279027d62cc.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libthiserror-cebe39f8c3e54657.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libhmac-519c517e9b74e166.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcrypto_mac-22c60dba3b034482.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libaes_ctr-d5a5a550793b9405.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libctr-18dd290096b381e7.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libaes_soft-6018baf98414cdfb.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcipher-ec40be554c9dcc71.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libsha2-75ce678b6df9a846.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcpuid_bool-ed065f766669b9b3.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libopaque_debug-b81c03213ec16149.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libblock_buffer-abe4ac60febb6928.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\librand-2099ab509b662c09.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\librand_chacha-fcbf1f0bdda88daa.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libppv_lite86-617c2dbdcaa3b30d.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libhkdf-5e22431a56356815.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libhmac-7eacb58a2d12a71b.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcrypto_mac-a4c89d03545e5fbb.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libdigest-ef89ad212bb6e9d8.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libgeneric_array-ba9d0da38ac4b07d.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libprost-902e37477f2cbffa.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libbytes-0c5b701a5b048f9a.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libx25519_dalek-2c45ffce1483d0c8.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcurve25519_dalek-6c765d3bf5d9acff.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libsubtle-4c476c8ec3b8559f.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libzeroize-92b5a284b0d5c763.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\librand_core-6abb38f769da791e.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libgetrandom-ff0dd3aa737217c3.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libdigest-de83356e5cae0a2e.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libgeneric_array-1d29f4de1cb1a3a2.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libtypenum-3695c9289495ef66.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libbyteorder-29142b0ab8bd55a4.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libstatic_assertions-0abfed3606d79f43.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\liblog-cdd0cfede6239e7b.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcfg_if-ad7ef0d6cbae1fed.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures-104ae9cbd5723bde.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_executor-7d3510d420977ff8.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_util-b9e8ab39d0483d85.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libmemchr-0e660e3abdafe1d1.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libproc_macro_nested-5f39e20fbfe09011.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_io-90ce7ac3aa792b1b.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libslab-7bc94dc3b903c652.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_channel-832d675866ac38ca.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libpin_project-7c93e5e3c7bde49f.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_sink-cf36a805deb031be.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_task-f777fe1f46435374.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libonce_cell-19ed83ac2e2d0b8d.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libpin_utils-82cde187427fa1a3.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfutures_core-c552bdeea4b6dc72.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libfailure-04fd84390c860ddd.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libbacktrace-b6af707aa65a01c3.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\libcfg_if-709f3978065f6469.rlib" "C:\\Users\\Dennis\\repos-c\\ringrtc\\src\\rust\\target\\i686-pc-windows-msvc\\release\\deps\\librustc_demangle-69b9d79268602572.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd-f4c158dab33a3977.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libpanic_unwind-a37722c46f2a18b4.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_demangle-1c6d276b48ed2ce8.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libhashbrown-9a87a105949005d7.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-78a58b3ad1e3d735.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libunwind-02743839f37c6208.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcfg_if-b228f18e39d8285c.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liblibc-1b403106181c3862.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc-8342f74face2334c.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_std_workspace_core-d12772f22e8379ca.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcore-81bf76d233082673.rlib" "C:\\Users\\Dennis\\.rustup\\toolchains\\1.47.0-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcompiler_builtins-90bdc0d502053891.rlib" "advapi32.lib" "cfgmgr32.lib" "gdi32.lib" "kernel32.lib" "msimg32.lib" "opengl32.lib" "user32.lib" "winspool.lib" "advapi32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "msvcrt.lib" "-s"
  = note: LINK : warning LNK4044: unrecognized option '/s'; ignored
             Creating library C:\Users\Dennis\repos-c\ringrtc\src\rust\target\i686-pc-windows-msvc\release\deps\ringrtc.dll.lib and object C:\Users\Dennis\repos-c\ringrtc\src\rust\target\i686-pc-windows-msvc\release\deps\ringrtc.dll.exp
          msvcrt.lib(chandler4gs.obj) : error LNK2019: unresolved external symbol __except_handler4_common referenced in function __except_handler4
          C:\Users\Dennis\repos-c\ringrtc\src\rust\target\i686-pc-windows-msvc\release\deps\ringrtc.dll : fatal error LNK1120: 1 unresolved externals


error: aborting due to previous error

error: could not compile `ringrtc`.

To learn more, run the command again with --verbose.
make: *** [Makefile:74: electron] Error 101
```

However, when I remove the static linking logic from `build.rs`, it works as expected.

My knowledge of static/dynamic linking is very limited, but could it be that an "all or nothing" approach is needed? So either compile everything statically through `-C target-feature=+crt-static`, or dynamically by removing `cargo:rustc-link-lib=libcmt`? What would you advise?

Reply:
For me both of those settings are needed, and there is no way to set these flags in build.rs, so it seems okay.